tmail-backend/jmap/extensions-api/src/main/java/com/linagora/tmail/james/jmap/contact/DomainContactProvisionerListener.java

```diff
@@ -0,0 +1,129 @@
+import org.apache.james.core.Username;
+import org.apache.james.events.Event;
+import org.apache.james.events.EventListener;
+import org.apache.james.events.Group;
+import org.apache.james.mailbox.events.MailboxEvents.MailboxAdded;
+import org.apache.james.mailbox.events.MailboxEvents.MailboxDeletion;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Mono;
+
+public class DomainContactProvisionerListener implements EventListener.ReactiveGroupEventListener {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DomainContactProvisionerListener.class);
+    private static final String IGNORED_DOMAINS_PROPERTY = "ignoredDomains";
+
+    public static class DomainContactProvisionerListenerGroup extends Group {
+
+    }
+
+    static final Group GROUP = new DomainContactProvisionerListenerGroup();
+
+    private final EmailAddressContactSearchEngine contactSearchEngine;
+    private final List<Domain> ignoredDomains;
+
+    @Inject
+    public DomainContactProvisionerListener(EmailAddressContactSearchEngine contactSearchEngine,
+                                            HierarchicalConfiguration<ImmutableNode> configuration) {
+        this.contactSearchEngine = contactSearchEngine;
+        this.ignoredDomains = extractIgnoredDomains(configuration.getString(IGNORED_DOMAINS_PROPERTY, ""));
+    }
+
+    private List<Domain> extractIgnoredDomains(String domains) {
+        return Arrays.stream(domains.split(","))
+            .map(Domain::of)
+            .toList();
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public boolean isHandling(Event event) {
+        return event instanceof MailboxAdded
+            || event instanceof MailboxDeletion;
+    }
+
+    @Override
+    public Publisher<Void> reactiveEvent(Event event) {
+        if (event instanceof MailboxAdded) {
+            return handleDomainContactAdded((MailboxAdded) event);
+        }
+        if (event instanceof MailboxDeletion) {
+            return handleDomainContactDeletion((MailboxDeletion) event);
+        }
+        return Mono.empty();
+    }
+
+    private Publisher<Void> handleDomainContactAdded(MailboxAdded event) {
+        if (event.getMailboxPath().isInbox()) {
+            Username username = event.getUsername();
+            Optional<Domain> maybeDomain = username.getDomainPart();
+            return maybeDomain.map(domain -> addDomainContact(domain, username))
+                .orElse(Mono.empty());
+        }
+        return Mono.empty();
+    }
+
+    private Mono<Void> addDomainContact(Domain domain, Username username) {
+        if (ignoredDomains.contains(domain)) {
+            return Mono.empty();
+        }
+        return Mono.fromCallable(() -> new ContactFields(username.asMailAddress(), "", ""))
+            .flatMap(contactFields -> Mono.from(contactSearchEngine.index(domain, contactFields)))
```

Comment:
I propose to do better: lookup the firstname and surname for the user in LDAP.

That is what I am doing in https://github.com/linagora/tmail-backend/issues/1946

Reply:
> the final decision on the ticket?

In the ticket description: `Similar than https://github.com/linagora/tmail-backend/issues/1946 but for domain contacts`

The other ticket used LDAP to get information. I was surprised to see that this does not need to use LDAP TBH.